src/parsers/component-parser.js

```diff
@@ -155,6 +155,10 @@ function parseComponent(
     template = templateTag[1];
   }
   if (options && !options.skipHTMLHash) {
+    // ignore adding scope hash
+    if (styleMatches || !styleMatches) {
+      return;
+    }
     template = prefixHtmlTags(template, componentHash);
```

Comment:
styleMatches is an array of all the `<style>` tags. so when there are no style tags, it will be an empty array which we can check by checking if the length is zero. `styleMatches.length <= 0`

> - <style> tag does not exist
This will solve the second case as mentioned.


To fix the first case,
> - `<style>` tag exists and has global attribute

We have to loop through styleMatches array and check if all of the style tags have `global` attribute in them.

We can do that with `.every` method of JavaScript: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every

So maybe we can do something like:

```suggestion
  const isStyleGlobal = styleMatches.length <= 0 || styleMatches.every(styleMatch => styleMatch.attributes.global === true))
  if ((options && !options.skipHTMLHash) || isStyleGlobal) {
    // ignore adding scope hash
    template = prefixHtmlTags(template, componentHash);
```